### PR TITLE
Add codespell pre-commit and fix typos

### PR DESCRIPTION
**Pull request by jarrodmillman**

### CODE REVIEW

File: static/teams/bots-graphs.html [hunk 1]
```diff
@@ -1,173 +0,0 @@
-<div class="chart" id="keen-example-chart"></div>
```
Comment (bsipocz) [inline static/teams/bots-graphs.html:1]: The bot will eventually moved here I think, so do this page really need to go in this PR?

File: static/teams/bots-graphs.html [hunk 2]
```diff
@@ -1,173 +0,0 @@
...
-  client
-    .query("count", {
-      event_collection: "backport_stats",
-      filters: [
-        {
-          operator: "not_contains",
-          property_name: "slug",
-          property_value: "test1",
-        },
-        { operator: "ne", property_name: "reason", property_value: "Sucess" },
```
Comment (jarrodmillman) [inline static/teams/bots-graphs.html:160]: `/Sucess/Sucess/`

File: static/teams/bots-graphs.html (hunk 1 above)
Comment (jarrodmillman) [reply static/teams/bots-graphs.html:1]: There is a typo in the file and https://scientific-python.org/teams/bots-graphs.html seems to be broken.

If we leave `static/teams/bots-graphs.html` in then we need to either
(a) fix the typo (but I can't see how it impacts the website),
(b) add `sucess` to the list of skip words, or
(c) tell codespell to ignore  `static/teams/bots-graphs.html`.

I am happy with any of the choices.

Comment (jarrodmillman) [reply static/teams/bots-graphs.html:1]: Also if we leave it in, we should move it out the team gallery directory. But I wasn't sure where it should go.

File: static/teams/bots-graphs.html (hunk 2 above)
Comment (bsipocz) [reply static/teams/bots-graphs.html:160]: This is code, I would add the file to the exceptions as the value might be typoed upstream if something is looking for this property value.

File: static/teams/bots-graphs.html (hunk 1 above)
Comment (bsipocz) [reply static/teams/bots-graphs.html:1]: I would ignore the file in this PR, and bring in @Carreau about bot convos separately.

Comment (jarrodmillman) [reply static/teams/bots-graphs.html]: Done!

**Pull request merged**